packages/template-retail-react-app/app/hooks/use-shopper-context-search-params.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useLocation} from 'react-router-dom'
+
+// Constants
+import {
+    SHOPPER_CONTEXT_SEARCH_PARAMS,
+    SHOPPER_CONTEXT_FIELD_TYPES
+} from '@salesforce/retail-react-app/app/constants'
+
+/**
+ * This hook will return a shopper context object when search params related
+ * to shopper context are present.
+ *
+ * @returns {Object} A shopper context object that can be passed to the Shopper Context API
+ */
+export const useShopperContextSearchParams = () => {
+    const {search} = useLocation()
+    const searchParamsObj = new URLSearchParams(search)
+    const {
+        geoLocation: geoLocationSearchParams,
+        customQualifiers: customQualiferSearchParams,
+        assignmentQualifiers: assignmentQualifierSearchParams,
+        ...rootQualifierSearchParams
+    } = SHOPPER_CONTEXT_SEARCH_PARAMS
+
+    const rootQualifiers = getShopperContextFromSearchParams(
+        searchParamsObj,
+        rootQualifierSearchParams
+    )
+    const geoLocation = getShopperContextFromSearchParams(searchParamsObj, geoLocationSearchParams)
+    const customQualifiers = getShopperContextFromSearchParams(
+        searchParamsObj,
+        customQualiferSearchParams
+    )
+    const assignmentQualifiers = getShopperContextFromSearchParams(
+        searchParamsObj,
+        assignmentQualifierSearchParams
+    )
+
+    return {
+        ...rootQualifiers,
+        ...(Object.keys(geoLocation).length && {geoLocation}),
+        ...(Object.keys(customQualifiers).length && {customQualifiers}),
+        ...(Object.keys(assignmentQualifiers).length && {assignmentQualifiers})
+    }
+}
+
+/**
+ * Converts search parameters into a shopper context object based on a provided mapping.
+ *
+ * @param {URLSearchParams} searchParamsObj - The search parameters object
```

Comment:
Minor: although this `getShopperContextFromSearchParams()` is exported, it's only meant for private use only (e.g. for testing).

```suggestion
 * @private
 * @param {URLSearchParams} searchParamsObj - The search parameters object
```